<head>

<meta name="description" content="">

<meta name="keywords" content="">

<meta name="author" content="">





</head>

<body bgcolor="#ffffff" topmargin="0" leftmargin="0" marginheight="0" marginwidth="0">





<table border="0" cellpadding="0" cellspacing="0" align="right">

<tr>

<td><!--***************************************************************************************-->

<A onmouseover="window.status&#13;&#10;='Homepage';

 return true"

 onmouseout ="window.status=''; return true" href="/~neill/" >

Neil's homepage</A>&nbsp;&nbsp;<font color="#660099">|</font>&nbsp;&nbsp; <!--***************************************************************************************-->



</td>



<td><!--***************************************************************************************-->

<A onmouseover="window.status&#13;&#10;='Research Group';

 return true"

 onmouseout ="window.status=''; return true" href="http://intranet.cs.man.ac.uk/mlo/" >

research group</A>&nbsp;&nbsp;<font color="#660099">|</font>&nbsp;&nbsp; <!--***************************************************************************************-->

</td>





<td><!--***************************************************************************************-->

<A onmouseover="window.status&#13;&#10;='People

 in the group'; return true"

 onmouseout ="window.status=''; return true" href="http://intranet.cs.man.ac.uk/mlo/people/" >

MLO Researchers</A>&nbsp;&nbsp;<font color="#660099">|</font>&nbsp;&nbsp; <!--***************************************************************************************-->

</td>



</tr>

</table>



<table border="0" cellpadding="0" cellspacing="0" width="436">



<tr>

<td width="47" valign="top"><!--***************************************************************************************-->&nbsp; 

<!--***************************************************************************************-->

</td>

<td><!--***************************************************************************************-->

<a href="http://www.manchester.ac.uk/"

 onMouseOver 

="window.status='image1'; return true"

 onMouseOut="window.status=''; return true" 

      align="right" valign  ="top" 

     ; <P 

     ><IMG alt="University Logo" title="Manchester Est. 1824" src="http://www.cs.manchester.ac.uk/ai/pictures/icons/est1824.gif"> <!--***************************************************************************************-->

</td></tr></table></body>

<head>

   <link href="http://intranet.cs.man.ac.uk/css/_print.css"

    type="text/css" rel="stylesheet" media="print" />



    <link href="http://intranet.cs.man.ac.uk/css/_base.css"

    type="text/css" rel="stylesheet" media="screen" />

    <style type="text/css" media="screen">

    /*<![CDATA[*/

    <!--

    @import url(http://intranet.cs.man.ac.uk/css/_layout.css);

    @import url(http://intranet.cs.man.ac.uk/css/_navigation.css);

    @import url(http://intranet.cs.man.ac.uk/css/_forms.css);

    @import url(http://intranet.cs.man.ac.uk/css/_boxes.css);

    -->

    /*]]>*/



    .currentsection  {

	 background-color: #f0f0ff; 

	 color:#000;

	 display: block; 

	 font-weight:normal;

	 padding: 4px 4px 5px 4px;

	 text-decoration:none; 

    }

    .multilevel-linkdiv-0 {

	 background-color: #f0f0ff; 

	 margin-left: 1.0em;

         text-align: left

    }

    a.greylink:link    { color: #c0c0c0; }

    a.greylink:visited { color: #d0d0d0; }

		

    .lined {

	 border-style: solid;

	 border-color: #006073;

	 border-bottom-width: 0px;

	 border-top-width: 1px;

	 border-left-width: 0px;

	 border-right-width: 0px;

    }

    table, caption {

      font-size: inherit;

      font-weight: inherit;

      font-style: inherit;

      font-variant: inherit;

    }

    </style>



    <style type="text/css" media="screen">

    /*<![CDATA[*/

    <!--

    .section {margin-left: 4em}

.abstract    { font-size:small }

span.author  { font-weight: bold }

span.regular {  }

span.journal {  font-style: italic}

span.papertitle     { font-style: italic; font-weight: bold }

    body { background-image: url(http://intranet.cs.man.ac.uk/ai//pictures/icons/logoleft.gif); }

    div.locator { background-color: #6666cc; background-image: url(); background-repeat: no-repeat; border: none; margin: 1px 0 0 0; padding: 0; width: 184px; height: 92px; }

    -->

    /*]]>*/

    </style>

<!--  End of Style Sheet CSS -->

</head>





<head>

<title>Single Input Motif Module Models with Gaussian Processes - MATLAB Software</title>

</head>



<body><div class="section">



<h1>GP Single Input Motif Module Software</h1>



<p>This page describes examples of how to use the Gaussian process single input

motif module modelling software (GPSIM) available for <a

href="http://www.cs.man.ac.uk/neill-bin/software/downloadForm.cgi?toolbox=gpsim">download

here</a>.



<p>If you are looking for general Gaussian process implementations please check

 <a href="http://www.cs.man.ac.uk/~neill/gpsoftware.html">here</a>.



<h3>Release Information</h3>




    <p>The GPSIM software can be downloaded
    <a href="http://www.cs.man.ac.uk/neill-bin/software/downloadForm.cgi?toolbox=gpsim">here</a>.
    <h2>Release Information</h2>
    <p><b>Current release is 0.1211</b>.
    <p>As well as downloading the GPSIM software you need to obtain the toolboxes specified below. <b>These can be downloaded using the <i>same</i> password you get from registering for the  GPSIM software.</b>
    <table>
    <tr>
    <td width="65%"><b>Toolbox</b></td>
    <td width="35%"><b>Version</b></td>
    </tr><tr><td><a href="http://www.cs.man.ac.uk/~neill/netlab/downloadFiles/vrs3p3">NETLAB</a></td><td> 3.3</td></tr>
<tr><td><a href="http://www.cs.man.ac.uk/~neill/optimi/downloadFiles/vrs0p132">OPTIMI</a></td><td> 0.132</td></tr>
<tr><td><a href="http://www.cs.man.ac.uk/~neill/ndlutil/downloadFiles/vrs0p161">NDLUTIL</a></td><td> 0.161</td></tr>
<tr><td><a href="http://www.cs.man.ac.uk/~neill/mltools/downloadFiles/vrs0p1311">MLTOOLS</a></td><td> 0.1311</td></tr>
<tr><td><a href="http://www.cs.man.ac.uk/~neill/kern/downloadFiles/vrs0p221">KERN</a></td><td> 0.221</td></tr>
<tr><td><a href="http://www.cs.man.ac.uk/~neill/minimize/downloadFiles/vrs0p2">MINIMIZE</a></td><td> 0.2</td></tr>
</table>



Minor bug fix in demBarenco1.m



<h4>Release 0.121</h4>



Minor fixes of the code for dealing with white noise handling in multiKern.



<h4>Release 0.12</h4>



<p>Release with scripts to recreate results in prepared journal paper.



<h4>Release 0.11</h4>



<p>This follow up release was for the NIPS paper.



<h4>Release 0.1</h4>



<p>This was the first release and included software used to create preliminary

results to be submitted to the Gaussian Processes in Practice Workshop.

<p>The toolboxes required to run the GPSIM toolbox are listed below.</p>



<h3>Examples</h3>



<h4>Linear Solution</h4>



<p>For the moment there are two examples of the linear solution given in

the toolbox. They are <code>demBarenco1</code> and

<code>demBarenco2</code>. They fit a Gaussian process model to the

single input module associated with p53. <code>demBarenco1</code> used

the pre-processed data by PUMA package which shows the results

displayed in the ECCB submitted paper Figure 1. <code>demBarenco2</code> used

the pre-processed data by MAS5 method which is exactly same as the one

Barenco used in their paper. The original work is described in <a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Barenco:ranked06&printAbstract=1">Barenco

<i>et al.</i></a>.



<p>To run the example write



<p><code>&gt;&gt; demBarenco1 </code>



<p>At the first time of running, it will take some time to load in the

data from the excel spreadsheet files. The excel spreadsheets contain

mmgMOS processed versions of the .cel files. This gives us a standard

deviation associated with the expression level as well as the

expression level itself.  The .cel files are available on <a

href="http://www.ebi.ac.uk/arrayexpress/">ArrayExpress</a>. Since they

are in log space, they must first be transfered to non-log space. Once

this process has been done, relevant profiles are extracted they are

saved in a file called

<code>./data/barencoPUMAdata.mat</code>. (Similarly, the associated

data file for MAS5 pre-processing are saved in a file called

<code>./data/barencoMASdata.mat</code>.) The variances and means

computed are therefore the variance and mean of a log normal

distribution. Obviously a Gaussian process can go negative but because

the expressions are not in log space they can't go negative. This is

clearly a weakness with this model.



<p>Once the data has been extracted, conjugate gradient optimisation of the

model commences. The first batch of numbers are a gradient check (the third

column should be zeros).



<p>The model that is being created is actually three models, one

Gaussian process for each repeat that is in the data. The covariance

parameters are shared though, this is a form of empirical hierarchical

Bayes. The implicit assumption is that each repeat is generated by an

independent sample of f(t) from a Gaussian process with the

<i>same</i> covariance parameters. This seems reasonable given the

biological experiments were from different .cel lines, <i>i.e.</i>

they were not technical replicates. However <a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Barenco:ranked06&printAbstract=1">Barenco

<i>et al.</i></a> seem to treat the data as if there is one f(t) for

all replicates. We also tried this, and results (in terms of estimated

degradation, sensitivity and basal transcription) were not

significantly different.



<p>We follow <a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Barenco:ranked06&printAbstract=1">Barenco

<i>et al.</i></a> in fixing the degradation rate of <i>p21</i> to 0.8

hr<sup>-1</sup> and the sensitivity to 1.0.



<p><center><img src="./demBarenco1_basal.png" width="60%"><br>

<img src="./demBarenco1_sensitivity.png" width="60%"><br>

<img src="./demBarenco1_decay.png" width="60%">

<p>Bar charts of (left) basal transcription rate, (middle) sensitivity and

(right) degradation rate. Black bars are results from <a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Barenco:ranked06&printAbstract=1">Barenco

<i>et al.</i></a>. White

bars are from our <code>demBarenco1</code>.</center>



<p><center><img src="./demBarenco1_profile1.png" width="60%"><br>

<img src="./demBarenco1_profile2.png" width="60%"><br>

<img src="./demBarenco1_profile3.png" width="60%"><br>

<p>Plots of the profiles predicted for each of the three replicates given in the

experiments. Solid lines are mean predictions, dashed lines are two standard

deviations of error, crosses are the points predicted from <a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Barenco:ranked06&printAbstract=1">Barenco

<i>et al.</i></a> for

the first replicate (scaled to match each profile).</center>



<h3>MAP Approximation for Non-Linear Response</h3>



<p>There are three examples of the use of the MAP approximation, the first is

<code>demBarencoMap1</code> which is the experiment shown in the

Figure 2 in the ECCB submitted paper. We implemented the Michaelis

Menten kinetics for the p53 data in this experiment.



<p>To run the example write



<p><code>&gt;&gt; demBarencoMap1 </code>



<p><center><img

src="./demBarencoMapMLPAct3Mlpmulti_profile1_slide.png"

width="60%"><br> <img

src="./demBarencoMapMLPAct3Mlpmulti_profile2_slide.png"

width="60%"><br> <img

src="./demBarencoMapMLPAct3Mlpmulti_profile3_slide.png"

width="60%"><br> <p>Plots of the profiles predicted for each of the

three replicates given in the experiments. Solid lines are mean

predictions, dashed lines are two standard deviations of

error. </center>



<p>The second is <code>demBarencoMap2</code> which is a simple

exponential response with an MLP style kernel as a prior over the

transcription factor concentration, and the third is

<code>demBarencoMap3</code> which is a recreation of the linear

experiment given above, made as a sanity check.



<h4>Repression Solution</h4>



<p>For the moment there is only one example of the repression case

given in the toolbox. It is <code>demEcoliMap1</code> which fits a

Gaussian process model to the single input module associated with

E. coli data set. <code>demEcoliMap1</code> used the same microarray

data as <a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Khanin:repression06&printAbstract=1">Khanin <i>et al.</i>, 2006</a>. They identify 14 target genes which are

repressed by the TF LexA and the mRNA measurement for these genes are

collected over six time points.



<p>To run the example write



<p><code>&gt;&gt; demEcoliMap1 </code>



<p><center><img

src="./demMapFullEcoliOptimInit4Translatemulti_profile1_slide.png"

width="60%"><br> <p>Plots of the predicted TF profile. Solid lines are

mean predictions, dashed lines are two standard deviations of

error.</center>



<p><center><img src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene1.png" width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene2.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene3.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene4.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene5.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene6.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene7.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene8.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene9.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene10.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene11.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene12.png"

width="60%"><br>

<img

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene13.png"

width="60%"><br>

<img 

src="./demMapFullEcoliOptimInit4Translatemulti_ExprsProfile_Rep1_Gene14.png"

width="60%">



<p>Predicted expression levels for all 14 target genes. Parameters

shown were obtained by maximum likelihood. Standard errors were not

available for this two-dye microarray data. There is no replica and

therefore a gene-specifec noise variance parameter was estimated for

each target gene. Data and reconstructed profiles are given in the

original scale. Times is measured in minutes.  </center>



<h4>Cascaded Differential Equations</h4>



<p>The last example shows a simple cascade of differential

equations. We applied the driven input model to a simple cascade in

Drosophila mesoderm development, focusing on the transcription factor

Mef2 (Myocyte enhancing factor 2).  We selected six targets of Mef2

identified by Chromatin immunoprecipitation (ChIP-chip) assays [<a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Sandmann:temporal06&printAbstract=1">Sandmann

<i>et al.</i>, 2006</a>] and which were observed to be up-regulated

after Mef2 is expressed. Affymetrix time course microarray data from

wild-type embryos [<a

href="http://www.cs.man.ac.uk/neill-bin/publications/bibpage.cgi?keyName=Tomancak:systematic02&printAbstract=1">Tomancak

<i>et al.</i>, 2002</a>] provides us with observations of Mef2

expression (y(t)) and the expression of the target genes xj (t) at

hourly intervals. The microarray time course was replicated three

times, and we used all three replicates to train the model

parameters. The raw data was downloaded from

<a href="ftp://ftp.fruitfly.org/pub/embryo_tc_array_data/">ftp://ftp.fruitfly.org/pub/embryo_tc_array_data/</a>. the relevant profile

(including selected target genes' names, expression levels and

standard errors) for generating the results shown in the submitted

paper are saved in a file called <code>./data/mef2Data.mat</code>.



<p>To run the example write



<p><code>&gt;&gt; demGpdisimMef2 </code>



<p><center> 

<img src="./demMef2Dros2TF_profile_Rep1.png" width="60%"><br> 

<img src="./demMef2Dros2TF_profile_Rep2.png" width="60%"><br> 

<img src="./demMef2Dros2TF_profile_Rep3.png" width="60%"><br> <p>Plots of

the predicted TF profiles for three replicas. Solid lines are mean

predictions, dashed lines are two standard deviations of

error.</center>



<p><center><img src="./demMef2Dros2_ExprsProfile_Rep1_Gene1.png"

width="60%"><br>

<img src="./demMef2Dros2_ExprsProfile_Rep2_Gene1.png"

width="60%"><br>

<img src="./demMef2Dros2_ExprsProfile_Rep3_Gene1.png"

width="60%"><br>

<p>Plots of the predicted driving input Mef2 mRNA profiles for three

replicas. Solid lines are mean predictions, dashed lines are two

standard deviations of error.</center>



<p><center><img src="./demMef2Dros2_ExprsProfile_Rep1_Gene2.png" width="60%"><br>

<img

src="./demMef2Dros2_ExprsProfile_Rep1_Gene3.png"

width="60%"><br>

<img

src="./demMef2Dros2_ExprsProfile_Rep1_Gene4.png"

width="60%"><br>

<img

src="./demMef2Dros2_ExprsProfile_Rep1_Gene5.png"

width="60%"><br>

<img

src="./demMef2Dros2_ExprsProfile_Rep1_Gene6.png"

width="60%"><br>

<img

src="./demMef2Dros2_ExprsProfile_Rep1_Gene7.png"

width="60%"><br>

<p>Predicted expression levels for all 6 target genes for one replica

as examples.

</center>



</div></body>









<p><center>Page updated on Tue May 19 11:09:49 2009</center><!-- footer -->



<body>

    <div id="footer">

      <div class="links">

        <p>

          | 

          <a href="http://www.manchester.ac.uk/aboutus/documents/disclaimer/"

             title="Disclaimer">Disclaimer</a> |

             <a href="http://www.manchester.ac.uk/aboutus/documents/privacy/"

             title="Privacy">Privacy</a> |

             <a href="http://www.manchester.ac.uk/aboutus/documents/copyright/"

             title="Copyright Notice">Copyright notice</a> |

             <a href="http://www.manchester.ac.uk/aboutus/documents/accessibility/"

             title="Accessibility">Accessibility</a> |

             <a href="http://www.manchester.ac.uk/aboutus/documents/foi/"

             title="Freedom of information">Freedom of information</a> | 

             <a href="http://www.manchester.ac.uk/aboutus/contact/feedback/"

             title="Feedback">Feedback</a> |

        </p>



      </div>

      <div class="pagestatus">

        <p>Please contact

          <a href="mailto:webmaster.cs@manchester.ac.uk">webmaster.cs@manchester.ac.uk</a>

          with comments and suggestions

        </p>

      </div>

    </div>



  </body>

