%% PNAStwoS.tex
%% Sample file to use for PNAS articles prepared in LaTeX
%% For two column PNAS articles
%% Version: Apr 15, 2008 
 

%% BASIC CLASS FILE
\documentclass{pnastwo}

%% ADDITIONAL OPTIONAL STYLE FILES
\usepackage{graphicx}
\usepackage{pnastwoF}
\usepackage{amssymb,amsfonts,amsmath}
\usepackage{subfig}
\usepackage{bm}
%\usepackage[comma,sort&compress]{natbib}
\usepackage{cite}
\renewcommand\citeleft{(} \renewcommand\citeright{)}

%% OPTIONAL MACRO DEFINITIONS
\def\s{\sigma}
\newcommand{\erf}{\operatorname{erf}}


%%%%%%%%%%%%
%% For PNAS Only:
\url{www.pnas.org/cgi/doi/10.1073/pnas.0709640104}
\copyrightyear{2008}
\issuedate{Issue Date}
\volume{Volume}
\issuenumber{Issue Number}
%\setcounter{page}{2687} %Set page number here if desired
%%%%%%%%%%%%

\begin{document}

\title{A model-based method for transcription factor target
  identification with limited data}

\author{Antti Honkela\affil{1}{Department of Information and Computer
    Science, Aalto University School of Science and Technology,
    Helsinki, Finland},
  Charles Girardot\affil{2}{Genome Biology unit,
    European Molecular Biology Laboratory (EMBL), Heidelberg, Germany},
  E. Hilary Gustafson\affil{2}{}, Ya-Hsin Liu\affil{2}{}, Eileen
  E.M. Furlong\affil{2}{}, Neil D.
  Lawrence\affil{3}{School of Computer Science, University of
    Manchester, UK} \and Magnus Rattray\affil{3}{}}

\contributor{Submitted to Proceedings of the National Academy of Sciences
of the United States of America}

\maketitle

\begin{article}
\begin{abstract}
  We present a computational method for identifying potential targets of a
  transcription factor (TF) using wild-type gene expression time series
  data. For each putative target gene we fit a simple differential equation model
  of transcriptional regulation and the model likelihood serves as a
  score to rank targets. The expression profile of the TF is modeled as a sample from a Gaussian process prior
  distribution that is integrated out using a non-parametric
  Bayesian procedure. This results in a parsimonious model with relatively few
  parameters which can be applied to short time series data
  sets without noticeable over-fitting. We assess our method using
  genome-wide Chromatin Immunoprecipitation
  (ChIP-chip) and loss-of-function mutant expression data for two TFs, 
Twist and Mef2, controlling mesoderm development in {\it Drosophila}. Lists of top-ranked
  genes identified by our method are significantly enriched for genes
  close to bound regions identified in the ChIP-chip data and
  for genes that are differentially expressed in
  loss-of-function mutants. Targets of Twist display diverse expression profiles
  and in this case a model-based approach performs significantly
  better than scoring based on correlation with TF expression. Our
  approach is found to be comparable or superior to ranking based on
  mutant differential expression scores. Also, we show how integrating
  complementary wild-type spatial expression data can further improve
  target ranking performance.
\end{abstract}

\keywords{transcriptional regulation | gene expression | systems biology | Gaussian process inference}

\abbreviations{TF transcription factor; ChIP Chromatin Immunoprecipitation;}

\dropcap{T}ranscription factors are key nodes in the gene regulatory
networks that determine the function and fate of
cells. An important first step in uncovering a gene regulatory network
is the identification of target genes regulated by a specific
transcription factor (TF). A
common approach to this problem is to experimentally locate physical binding
of TF proteins to DNA sequence {\em in vivo} using a genome-wide
chromatin immunoprecipitation (ChIP)
experiment~\cite{Ren2000,Iyer2001}. However, recent studies suggest
that many observed binding events are neutral and do not regulate
transcription, while regulatory binding events often occur at
enhancers that are not proximal to the target gene that they
control~\cite{Li2008,MacArthur2009}. Therefore, the task of
identifying transcriptional
targets requires the integration of ChIP binding predictions
with evidence from expression data to help associate binding events
with target gene regulation. If there is access to expression data from a mutant in which the TF has
been knocked out or over-expressed, then differential expression of genes between wild-type
and mutant is indicative of a potential regulatory
interaction~\cite{Furlong2001,Sandmann2006a}. Available spatial expression data for the
TF and putative target can also provide support for a hypothesised
regulatory link. 

A problem with the above approach is that the creation of mutant
strains is challenging or impossible for many TFs of interest. Even
when available, mutants may provide very limited information because of redundancy or due to the
confounding of signal from indirect regulatory feedback~\cite{Gitter2009}. For these reasons it
is useful to seek other sources of evidence to complement ChIP binding
predictions. In this contribution we demonstrate how a dynamical model of wild-type transcriptional
regulation can be used for genome-wide scoring of putative target genes. All that is required to apply
our method is wild-type time series data collected over a period
where TF activity is changing. Our approach allows for complementary
evidence from expression data to be integrated with ChIP binding data
for a specific TF without carrying out TF perturbations. 

To score putative targets we use the data likelihood under a simple
cascaded differential equation model of transcriptional regulation. The regulation model
we apply is ``open'' in the sense that we do not explicitly model regulation of the TF
itself. To deal with this technical issue we use a recently developed
non-parametric probabilistic inference methodology to
effectively deal with open differential equation
systems~\cite{Gao2008}. We model the TF concentration as a function
drawn from a Gaussian process prior distribution~\cite{Rasmussen2006,Lawrence2010}.
This functional prior can either be placed
on the TF mRNA, for TFs primarily under transcriptional regulation,
or the TF protein, for TFs activated post-translationally. In the
application considered here the TFs are transcriptionally regulated
and we take the former approach. We use Bayesian marginalisation (also
known as Bayesian model averaging) to
integrate out these functional degrees of freedom. This greatly reduces the
number of parameters required to model the data, making a
likelihood-based approach feasible even for short time series.

There are many existing approaches to inferring gene regulatory networks from
time series expression data, including dynamic Bayesian networks,
information theoretic approaches and differential equation approaches
(reviewed in \cite{Bansal2007a}). These methods typically require many
more data from a greater diversity of experimental conditions than are
available from the short unperturbed wild-type time series that we
consider. Indeed, most real gene expression time course data are short
relative to the simulated data used to assess computational methods
for network inference~\cite{Ernst2005}. However, our goal is more limited in scope since
we are primarily interested in providing additional support for hypothesised
targets of a specific TF. Again, most approaches to this problem are
designed for data containing large numbers of diverse conditions, as
exemplified by the DREAM 2 target identification
challenge 1~\cite{Stolovitzky2007}. Others
have addressed this target identification problem using time series
data with a regulation model~\cite{Barenco2006a,Gatta2008}. However,
these approaches either require a known target set for training~\cite{Barenco2006a} or
they require measured TF protein data~\cite{Gatta2008}. In addition to
these differences in the assumed prior knowledge and available data,
it is also difficult to validate other approaches on the same data used in these
studies as they only carried out validation of selected targets
that they identified, rather than using unbiased genome-wide
experimental validation. 

We validate our proposed method by comparing the model-based target
ranking with published ChIP-chip data for two TFs controlling
mesoderm development in \emph{Drosophila}. Our method is shown to be comparable
to, or outperform, the use of knockout mutant data which are available
for these TFs. We demonstrate that a model-based approach is significantly
better than a simpler approach using correlation when targets display a diverse set of expression profiles. We further show how integrating complementary wild-type spatial
{\em in situ} expression data can greatly improve target ranking accuracy. 

\begin{figure*}[tb]
  \centering
  \includegraphics{fig1}
  \caption{Examples of the model fit for two different classes of 
    Gaussian process model fitted to potential targets of the TF Twist. (a) Three independent single-target models
    for likely targets. Red marks denote observed expression
    levels from~\cite{Tomancak2002} with 2 s.d.\ error bars.
    The inferred posterior means of the functions are shown in blue
    and the shaded regions denote 2 s.d.\ posterior confidence
    intervals. (b) A joint multiple-target model for
    the same set of target genes as in (a). Note that
    the multiple-target models used in evaluation have more targets
    than the one shown here.\label{fig:gpdisim_models}
}
  %\label{fig:gpdisim_models}
\end{figure*}

\section{Gaussian process inference for a linear activation model}

A Gaussian process is a probability distribution over
functions $f$ which take the value $f(t)$ at time
$t$. Analogous to the Gaussian distribution, which is fully
characterised by a mean and covariance, the Gaussian process is
characterised by a mean function $\mathrm{E}[f(t)]$ and a covariance
function $k(t,t')=\mathrm{E}[f(t)f(t')]-\mathrm{E}[f(t)]\mathrm{E}[f(t')]$
where the expectation is over function evaluations at times $t$ and
$t'$~\cite{Rasmussen2006}.

For the prior distribution of TF mRNA concentration profile $f$ we use a
zero-mean Gaussian process with squared exponential covariance
$k(t,t')=a\exp(-(t-t')^2/l^2)$. Samples from a Gaussian process with this choice
of covariance are smooth, infinitely differentiable, stationary functions. Although single-cell mRNA counts may be low
and vary in a highly stochastic manner, here we consider microarray data which
corresponds to a population average and is therefore more
appropriately modeled using a smooth process. The parameters $a$ and
$l$ describe the typical amplitude and time-scale of samples from
the prior. 

To simplify the inference we model translation and transcriptional
activation %as simple linear processes
using linear ordinary differential
equations,
\begin{align}
  \frac{\mathrm{d}p(t)}{\mathrm{d}t} & = f(t) - \delta
  p(t) \ , \label{eq:translation_ode} \\
  \frac{\mathrm{d}m_j(t)}{\mathrm{d}t} & = B_j+S_j p(t)-D_j m_j(t) \ , \label{eq:transcription_ode}
\end{align}
where $p(t)$ is the TF protein and $m_j(t)$ is the $j$th target mRNA
concentration at time $t$. The parameters $B_j$, $S_j$ and $D_j$ are the
baseline transcription rate, sensitivity and decay rate respectively
for the mRNA of the $j$th target (as described in \cite{Barenco2006a}).
The parameter $\delta$ is the decay rate of the TF protein.
This linear system of differential equations can be
explicitly solved and we find that the functions $p$ and $m_j$ are
linear functions of $f$. Given $k$ target genes the functions $f$ and $\bm m\equiv[m_j]$ therefore
describe a $k$+1-dimensional Gaussian process prior $p(f,\bm m|\theta)$
for which the covariance function can be calculated explicitly in terms of the
$3$ global and $3k$ gene-dependent model parameters
$\theta=[\delta,a,l,\{B_j,S_j,D_j\}]$ (see Materials and Methods). 

A microarray time course experiment provides a data vector $\bm y$ containing the noise-corrupted TF and target mRNA
measurements at $n$ times,
$\bm
y=[\hat{f}(t_1),\hat{f}(t_2),\ldots,\hat{f}(t_n),\{\hat{m}_j(t_1),\ldots,\hat{m}_j(t_n)\}]^\mathrm{T}$.
In this continuous-time modeling framework there is no
requirement that times are equally spaced. We assume Gaussian noise:
$p(\bm y|f,\bm m)=\prod_i{\cal
  N}(\hat{f}(t_i)|f(t_i),\sigma_{if}^2)\prod_j{\cal
  N}(\hat{m_j}(t_i)|m_j(t_i),\sigma_{ijm}^2)$ where the gene and condition
specific measurement variance parameters $\sigma_{if}^2$ and
$\sigma_{ijm}^2$ are obtained from a probe-level analysis of the
microarray data~\cite{Liu2005,Pearson2009} (see
Materials and Methods). The log-likelihood for the model parameters
$\theta$ can then be calculated exactly using standard Gaussian process regression
techniques to integrate out the functions $\bm m$ and
$f$~\cite{Rasmussen2006}. For independent replicate datasets, with
$R$ experiments in total, the
log-likelihood is
\begin{equation*}
  \begin{split}
    L(\theta) & = \log p(\bm y_1,\bm y_2,\ldots,\bm y_{R}|\theta) \\
    & = \sum_{r=1}^{R} \log \!\int 
    \!\!p(\bm y_r|f,\bm m)p(f,\bm m|\theta) \, \mathrm{d}\!f\mathrm{d}\bm m\\
    & = \sum_{r=1}^{R} \left[-\frac{1}{2}(\bm y_r - \bm \mu_y)^\mathrm{T} K_r^{-1} (\bm y_r - \bm \mu_y) -
      \frac{1}{2}\log|K_r|\right] -C      %nR\log 2\pi
  \end{split}
\end{equation*}
where $\bm \mu_y$ is the mean and $K_r$ is the covariance matrix for the
noise-corrupted dataset $\bm y_r$ under the Gaussian process prior (see
Materials and Methods), and $C$ is a constant independent of the
parameters.  This allows us to compute the set of model
parameters $\theta$ by maximum
likelihood using gradient-based optimisation. 

We use the likelihood to rank genes according to their fit to the
regulation model. Genes with a
near-constant profile may have high likelihoods without being targets,
but we find that filtering of weakly expressed genes removes most of
these (see Materials and Methods). It is also possible to filter
targets using a comparison to a model with
zero sensitivity but we find that the proposed approach achieves
similar results with significantly smaller computational cost.

Rather than assuming that the TF protein is primarily under
transcriptional control, as indicated by
Eq.~[\ref{eq:translation_ode}], one can alternatively use the idea pioneered
in~\cite{Barenco2006a,Khanin2006} and model the TF protein as an unobserved
latent variable. This approach can naturally be carried out using a
similar Gaussian process framework as shown in~\cite{Gao2008}.
However, 
unlike the approach described above,
that procedure requires a set of known
targets for training the model. %whereas the approach described above
%can be applied without any prior knowledge of targets.
We therefore
prefer the above cascade model for cases where the level of active
TF is under direct transcriptional control. 

\section{Results}

We applied our model-based approach to identify targets of the TFs
Twist and Mef2 which regulate mesoderm and muscle development in
\emph{Drosophila}~\cite{Sandmann2007,Zinzen2009}. We used a microarray dataset
containing three independent time series of 12 time points collected
hourly throughout \emph{Drosophila} embryogenesis in wild-type
embryos~\cite{Tomancak2002}.  Weakly expressed candidate targets were eliminated by 
considering average z-scores of expression values as described in
Materials and Methods.

For each TF we fitted two sets of Gaussian process models. In one
approach all other genes were considered as individual candidate targets one at a
time (``single-target GP'').  The candidate targets were ranked
according to their likelihood under the model.  Note that even the global
parameters can vary between models for different genes under this approach as the models
are fitted completely independently. In an alternative approach we fitted another set of models using the five
top-ranking genes in the single-target case as the training set (``multiple-target GP''; see
Materials and Methods).  An illustration comparing these alternative
modeling approaches applied to the TF Twist is shown in Fig.~\ref{fig:gpdisim_models}. The
single-target model is more flexible because each
potential target gene can be associated with a different inferred TF
protein profile. The multiple-target model is constrained to explain
all targets with the same TF protein profile and may therefore assign
a low score to targets that have a high score according to the
single-target model (this is the case for gene FBgn0003486 in
Fig.~\ref{fig:gpdisim_models}). Nevertheless, both models are
sufficiently flexible to fit target genes with significantly different
profiles (genes FBgn0033188 and FBgn0035257 in
Fig.~\ref{fig:gpdisim_models}).
In Fig.~\ref{fig:gpdisim_mef2_models} we show some examples of the
single-target model for Mef2.  For this TF most of the
top-ranking targets identified by the model are similar to the
profiles in the first two columns of the figure.
The predicted Mef2 targets show much less variability in expression
profile than those for Twist.

\begin{figure}[tb]
  \centering
  \includegraphics{fig2}
  \caption{Examples of the model fit for three independent single-target Gaussian process models for
    likely targets of the TF Mef2 (compare to the examples in Fig.~\ref{fig:gpdisim_models}(a) for Twist).\label{fig:gpdisim_mef2_models} }
\end{figure}

To evaluate the advantage from using Gaussian processes in our
approach, we performed a comparison to direct maximum likelihood fit
of the same differential equation model using the observed TF mRNA
levels $\hat{f}_i(t_j)$ in trapezoidal quadrature evaluation of
Eqs.~[\ref{eq:gpsim_f_ode_sol}--\ref{eq:gpsim_f_x_sol}]
(``single-target quadrature'').
For comparison, candidates were also ranked by correlation of the
expression profile with the TF expression profile (``correlation'')
and by $q$-value of differential expression in TF loss-of-function mutant
experiments from~\cite{Sandmann2006a,Sandmann2007} (``knock-out''). 
To assess the significance of the observed differences between
alternative ranking approaches, we performed 100,000-fold bootstrap
resampling of the dataset, 
recording relative performances of the different methods for each
fold. Full bootstrap results are in Table S1.
%, the Gaussian process methods were superior to the alternatives
%in over 95\% of the repeats in most Twist validation settings.
%For Mef2, there are fewer highly significant differences.


%The rankings of the targets shown using single-target models are 2, 27
%and 4910.  The corresponding rankings based on correlation are 241, 13
%and 5.

\begin{figure*}[tb]
  \centering
  %\includegraphics[trim=0mm 18mm 0mm 0mm]{dros_global_evaluation}
  \includegraphics{fig3}
  \caption{Evaluation results of different rankings as
    discussed in the text showing the relative frequency of positive
    predictions among $N$ top-ranking targets (``global'' evaluations)
    or among $N$ top genes
    with annotated expression in mesoderm or muscle tissue
    (``focused'' evaluations).
    The dashed line
    denotes the frequency in the full population and the dash-dot
    line within the population considered in focused evaluation.
    The first row shows the frequency of targets with ChIP-chip
    binding within 2000 base pairs of the gene
    while the second row shows the frequency of
    predicted targets with significant differential
    expression in TF knock-outs.
    $p$-values of results significantly different from random are
    denoted by `***': $p <
    0.001$, `**': $p < 0.01$, `*': $p < 0.05$.
    Comparison to knock-out ranking is obviously omitted for knock-out
    validation. \label{fig:dros_global_evaluation}
  }
 % \label{fig:dros_global_evaluation}
\end{figure*}

The accuracy of the ranking was first evaluated by looking globally at
the fraction of $N$ top-ranking targets that are significantly
differentially expressed in knock-outs of the TFs and the fraction
of targets with ChIP-chip binding within 2000 base
pairs of its beginning or end or within the gene in the data
from~\cite{Zinzen2009}. For Twist the single-target and
multiple-target models both perform well, with the single-target model
performing slightly better (first column of
Fig.~\ref{fig:dros_global_evaluation}). The simplified quadrature
method is almost as good. In the ChIP validation the
model-based methods both perform significantly better than ranking
using either mutant differential expression or correlation with TF
expression. In the knock-out validation all model-based approaches
significantly out-perform the correlation ranking. For Mef2
(third column of Fig.~\ref{fig:dros_global_evaluation}) the results of all
methods have more roughly comparable performance. In the ChIP validation all
methods provide highly significant enrichment in the top 100 and top
250 lists with the correlation performing best. In the knock-out
validation the model-based and correlation-based rankings both provide
highly significant enrichment in the top 100 and top 250 lists, with
the single-target model performing best.

% \begin{figure}[tb]
%   \centering
%   %\includegraphics[trim=0mm 8mm 0mm 0mm]{dros_focused_evaluation}
%   \includegraphics{fig4}
%   \caption{Evaluation results of different rankings based on
%     ChIP-chip data and differential expression in knock-outs
%     similar to those in
%     Fig.~\ref{fig:dros_global_evaluation} except only considering
%     $N$ top genes
%     with annotated expression in mesoderm or muscle tissue.
% %     as discussed in the text showing the relative frequency of
% %     positive predictions among $N$ top-ranking targets.  The dashed
% %     line denotes the frequency in full population.  The first row
% %     shows the frequency of predicted targets that are significantly
% %     differentially expressed in knock-outs for the TF.  The second row
% %     shows the frequency of targets with ChIP-chip binding within 5 kb
% %     of the gene.  $p$-values results significantly different from
% %     random are denoted by '***': $p < 0.001$, '**': $p < 0.01$, '*':
% %     $p < 0.05$.
% \label{fig:dros_focused_evaluation}
% }
%   %\label{fig:dros_focused_evaluation}
% \end{figure}

Because Twist and Mef2 activities are specific to the mesoderm and
muscle development, we also looked at accuracy focusing on genes
with annotated expression in these tissues according to the
\emph{in situ} data from~\cite{Tomancak2002}. These results are
illustrated in the second and fourth column of
Fig.~\ref{fig:dros_global_evaluation}. For the Twist
ChIP validation we now observe better performance for the knock-out
scores although the model-based rankings remain best. However, we
observe that the correlation ranking is no longer statistically
significantly enriched for ChIP binding on this spatially filtered set
of targets. For the Mef2 ChIP validation the relative performance of
all methods is similar to the global enrichment analysis
although the knock-out top 20 ranking is improved.
Overall, the Gaussian process methods tend to perform better than the
quadrature-based method. The differences are statistically significant
in focused Twist validation for top 250 list and several Mef2 validation
settings for top 100 and top 250 lists.

% The results for $N=10$ are only significantly better than random for
% Twist ChIP-chip, while for $N=50$ and especially $N=250$ the
% improvement is highly significant.  Especially for Twist the results
% are even now clearly better than any of the tested alternatives.

The distance threshold of 2000 bp used in ChIP-chip evaluation is
somewhat arbitrary.  The performance of different methods when this
threshold is changed is illustrated in
Fig.~\ref{fig:dros_binding_site_distances} and in Fig.~S1.
The figure shows that the relative performances of
different algorithms are fairly consistent for different choices of threshold. 
We see that the model-based rankings consistently perform best for Twist, while 
methods perform more similarly for Mef2 with the correlation and single-target models 
performing comparably well.

\begin{figure}[tb]
  \centering
  \includegraphics{fig5}
  \caption{The fraction of top 250 predicted targets having a ChIP-chip
    binding site (\emph{cis}-regulatory module; CRM)
    within $x$ base pairs as a function of $x$.
    Binding sites within an intron of a gene are considered to
    have a distance 0.
    \label{fig:dros_binding_site_distances}}
\end{figure}

% To assess the significance of the observed differences between
% alternative ranking approaches, we performed bootstrap
% resampling of the data set.  In 100,000-fold resampling where the
% relative performances of different methods were recorded for each
% fold, the Gaussian process methods were superior to the alternatives
% in over 95\% of the repeats in most Twist validation settings.
% For Mef2, there are fewer highly significant differences.
% (Full results in Supplementary Material.)

% We also repeated the experiment with the method of~\cite{Gatta2008}
% but using the TF expression levels as a proxy for the measured protein
% levels.  The results of this method are in general comparable to
% ranking by correlation; there are no cases where it would be
% significantly better than correlation and especially in cases where
% correlation is doing well it is often worse.
% Using measured protein levels as proposed in~\cite{Gatta2008} would
% naturally help, but unfortunately such information is not available in
% this case or in general.  It is worth noting that any measurements of
% protein levels could very easily be incorporated in our models as well.

Finally, since most expression time series are short~\cite{Ernst2005}
we assessed whether similar results could be obtained with fewer time
points. We sub-sampled the data keeping only 7 time points most
relevant in defining the Twist expression profile
(2,3,4,5,6,8,10 hours). The results in this case, as shown in
Fig.~\ref{fig:dros_data_size_evaluation}, are very similar to 
those obtained using the complete data showing that 
%the model-based approach presented here
our approach
can be applied to typical short wild-type time
series data.

\begin{figure}[tb]
  \centering
  \includegraphics{fig6}
  \caption{Evaluation results similar to those in 
    Fig.~\ref{fig:dros_global_evaluation}
    %and \ref{fig:dros_focused_evaluation}
    for single-target Gaussian process models
    for full data (12 time points) and sub-sampled data (7 time points).
    \label{fig:dros_data_size_evaluation}
  }
\end{figure}

\section{Discussion}

We have shown how a very simple regulation model can be used to rank
genes according to their likelihood under a model of transcriptional
regulation by a TF of interest. The linear differential equation model of regulation we use
is a simplification, and in many cases unrealistic, but it can capture important differences in
the profiles of target genes due to differences in protein and mRNA
stability while still keeping the computational load at a level
that allows us to perform genome-wide scoring using reasonably modest
computational resources without the need for expensive sampling.
When data are limited then the linear model can provide a sufficiently good
approximation to a non-linear model. 
The linear model does not restrict concentrations from being negative
and occasionally the inferred credible region of functions extends to negative
values. However, as the parameters and observations are constrained to be positive,
this is rare and does not seem to affect the ranking performance. Our
results show that the model-based approach presented here can provide
a viable alternative or complementary method to the use of
loss-of-function mutant data for identifying targets. Such an approach
can be combined with binding location (ChIP-chip or ChIP-seq)
and spatial expression data to identify a confident set of regulatory
targets. This is especially useful in cases where loss-of-function
mutants are uninformative, either due to redundancy or because they do
not display a phenotype of interest. 

The main novelty of our approach is the use of a differential equation
transcription model while minimising the number of free parameters.
The most effective method is based on recently proposed
techniques to model a time-varying degree of freedom as a sample from
a Gaussian process prior distribution~\cite{Gao2008}. By marginalizing
out these functional
degrees of freedom before computing the likelihood, we keep the
number of model parameters to a minimum and this allows us to model
typical short wild-type time series without noticeable over-fitting. In
many applications it is difficult or impossible to achieve a high
degree of temporal resolution. For example, in studies of
embryogenesis it is often impossible to stage embryos with the
precision sufficient to obtain a long expression time series over the
period of interest. For one reason or another most expression time
series datasets are short~\cite{Ernst2005} and developing inference approaches
that are applicable to short time series datasets is of great
practical importance.

Related methods have been proposed for identifying the
targets of a TF using time series expression data but few of these are suitable given the
limited data considered here~\cite{Bansal2007a}. Our approach was inspired by the work of Barenco {\em et
al.} who also use a differential equation with a linear activation
function to model the regulation of transcription. They focus on the
case where the TF is activated post-translation and they therefore
require the use of a training set of
known targets to infer the TF activity~\cite{Barenco2006a}. In the most recent
implementation of their method~\cite{Barenco2009} the original Markov
chain Monte Carlo procedure for Bayesian inference has been dropped due to its computational
inefficiency and an alternative optimisation-based method is used. In our case a Bayesian treatment of the TF protein
concentration is computationally efficient because we use Gaussian
process inference techniques.
%A Bayesian approach reduces the number of parameters
%to be optimised, since we marginalise over the TF concentration, and
%thereby reduces the possibility of over-fitting.
Nevertheless, the use of a training set to infer the
TF concentration from targets was a very elegant solution to the
problem of dealing with TFs activated post-translation. This
approach also fits very naturally into the Gaussian process
framework as we have shown~\cite{Gao2008}. Such a scheme was unnecessary in
this application,
where the TFs of interest were known to be transcriptionally
regulated, but can be applied using our techniques in other
contexts.
% Unfortunately it is difficult for us to compare our method
% directly with the results of Barenco {\em et al.} since their results
% were only validated using the positive and negative targets that they identified,
% rather than an unbiased test set. 

Another relevant method was introduced by Della Gatta {\em et
  al.}~\cite{Gatta2008}. They also use a linear regulation model but
they introduce simplifications that avoid the explicit solution of
differential equations. An interesting aspect of their approach is the
inclusion of a regulatory network model designed to capture the effect
of indirect regulation of putative targets by other genes. Sparse regression
techniques are used to learn the regulation model. A limitation of
their approach is the necessity to have measurements of the TF
protein; such data are not readily available in most cases. As with
Barenco's approach it is difficult for us to compare performance on
the data they used in~\cite{Gatta2008} because they only validated
selected predictions of their model. In our application we do not have
access to any protein data and therefore were only able to compare to
their approach using mRNA data as a proxy for protein data.
The results of this somewhat unfair comparison show significant
enrichment in only slightly more than a half of the evaluation
settings.
%  cannot apply their method as
% intended. We were interested in whether
% their approach to modelling indirect regulation would be helpful in
% our case. Therefore we applied their approach using mRNA data as a
% proxy for protein data which is not unreasonable for TFs under
% transcriptional control. The results only show significant
% enrichment in global evaluation for top 100 and top 250 and Twist
% focused evaluation for top 250 and are in most cases no better than
% the our models and correlation (see Fig.~S2 for full
% results). However, this comparison is
% not fair to their method which really relies on the
% availability of protein measurements.
It is worth noting that any measurements of protein levels to help
their model could very easily be incorporated in our models as well.

There are some interesting differences in the ranking results for the two TFs studied
here. The knock-out scores for Twist are found to be less informative
about binding location than the knock-out scores for Mef2. This may be
because Twist acts earlier and therefore has many more indirect
downstream targets in the regulatory network. Indeed, Mef2 is a
direct target of Twist and therefore the direct targets of Mef2 are
indirect targets of Twist. These indirect targets will be
differentially expressed in the mutant and will therefore increase the
false positive rate. We found that the model-based scores provide 
a significant improvement over correlation with TF expression for Twist. 
This is most likely due to the diversity of target gene expression profiles
 for this TF (recall Fig.~\ref{fig:gpdisim_models}). The models can fit 
profiles that are very poorly correlated with the TF expression profile. For Mef2
the models still provide highly significant enrichment of both ChIP and knock-out 
predictions, but the correlation performs similarly well. Fig.~\ref{fig:gpdisim_mef2_models} 
shows that predicted Mef2 targets are much less diverse in profile than those for Twist. Also, Mef2 is
active later, at stages downstream of Twist. Therefore, there
are less data points available to capture the diversity of profiles
that can be explained by the regulation model while the correlation with TF profile will typically be good. Conversely, the
correlation score is much less useful for Twist, since it misses many
targets that have profiles very different from the Twist mRNA profile. 

The two model-based approaches (single-target and multiple-target) perform 
similarly well, although the single-target approach gave slightly better results in 
most comparisons. Fig.~\ref{fig:gpdisim_models} suggests that the more constrained 
multiple-target model conforms more with our assumption of a shared protein 
profile for all targets. However, performance of this model may be strongly
influenced by the choice of training genes. In order not to bias our validation, 
we chose training genes using the single-target model ranking, without any reference
to ChIP-chip, {\it in situ} or knock-out data. In practice it may be better to use a more 
confident set of training genes. 

We are pursuing a number of extensions of the current work. Here, we
have limited ourselves to detecting the targets of transcriptional activators. The
Gaussian process approach can be used to compute the likelihood for
repression models~\cite{Gao2008}. However, since models of repression
are inherently non-linear~\cite{Khanin2006}, the likelihood calculation is no longer analytically
tractable. Therefore approximate Bayesian inference methods are
required and we are currently working on an efficient implementation
for genome-wide ranking. We have also limited our attention
to single input motif models, but many targets will be regulated by other TFs and
co-factors. It seems doubtful that more complex models of regulation
will be useful for target ranking using only short wild-type time
series data, since it is unlikely these data would be sufficient to parametrise more complex models. However,
more general regulation models may be applicable if we were to use more data for
inferring the model parameters, e.g. time series knock-out data and time series
ChIP data could be highly informative. Scoring multiple input
models moves us closer to the more general gene regulatory network inference
problem. We expect that Gaussian processes will provide a useful way
to deal with inference over time-varying external inputs in these and other
biological systems, thus paving the way for integration of
probabilistic and mechanistic modelling techniques.

\subsection{Availability}
Complete Matlab code for repeating the experiments is available at
\texttt{http://users.ics.tkk.fi/ahonkela/gpdisim/}.
An R implementation of
the ranking method is available from the authors upon request.

\begin{materials}
  \section{Expression data preprocessing} {\it Drosophila} developmental
  microarray time series from~\cite{Tomancak2002} were reprocessed
  using mmgMOS from the puma package~\cite{Pearson2009}.  The means of
  the log-scale expression values were equalized across chips.  The
  distributions over log-expression were transformed to means
  $\left(\hat{m}_i(t_j), \hat{f}_i(t_j)\right)$ and variances $\left(\sigma_{ijm}^2,
  \sigma_{if}^2 \right)$ of absolute expression by minimising the squared
  deviation of 5\%, 25\%, 50\%, 75\% and 95\% quantiles of a Gaussian
  distribution to corresponding quantiles obtained from mmgMOS.

  \section{Filtering weakly expressed genes}
  Weakly expressed genes can affect the ranking because any model
  with low or zero sensitivity can fit them
  well. Genes $j$ with an average z-score 
  $$ \frac{1}{Rn} \sum_{r=1}^R \sum_{i=1}^n \frac{\hat{m}_{jr}(t_i)}{\sigma_{ijr}} < 1.8 $$
  were considered weakly expressed and removed from the dataset.  The threshold
  was selected by visually inspecting models for candidate targets:
  practically all targets below the threshold elicited a
  non-informative constant response. % from the model.
  
  \section{The Gaussian process model}
  Under suitable initial conditions, the differential
  equations~[\ref{eq:translation_ode}--\ref{eq:transcription_ode}]
  can be solved to obtain
  \begin{align}
    \label{eq:gpsim_f_ode_sol}
    p(t) &= \exp(-\delta t) \int_0^t f(v) \exp(\delta v) dv\ , \\
    m_j(t) &= \frac{B_j}{D_j} + S_j \exp(-D_j t) \int_0^t p(u)
    \exp(D_j u) du\ .
    \label{eq:gpsim_f_x_sol}
  \end{align}
  Assuming $\mathrm{E}[f(t)]=0$, then $\mathrm{E}[p(t)]=0$ and
  $\mathrm{E}[m_j(t)]=\frac{B_j}{D_j}$.
  Using these solutions and assuming a squared exponential covariance
  for $f(t)$, $k_{ff}(t, t') = a \exp( -(t-t')^2/l^2)$, the covariance
  of the TF and target mRNAs is
  \begin{align*}
    k_{f m_k}(t, t')
    &= \frac{\sqrt{\pi} l a^2 S_k}{2(\delta - D_k)}
    \left(h(t, t', D_k) - h(t, t', \delta) \right),\\
    h(t, t', D) &= e^{\gamma_D^2 + D \Delta t}
     \left[\erf\left(\gamma_D + \frac{t}{l}\right) 
       - \erf\left(\gamma_D + \frac{\Delta t}{l}\right)\right] \\
%     \bigg\{
%     \exp\left(\gamma_k^2 + D_k \Delta t \right)
%     \left[\erf\left(\gamma_k + \frac{t}{l}\right) 
%       - \erf\left(\gamma_k + \frac{\Delta t}{l}\right)\right] \\
%     -
%     \exp\left(\gamma_\delta^2 + \delta \Delta t\right)
%     \left[\erf\left(\gamma_\delta + \frac{t}{l}\right) 
%       - \erf\left(\gamma_\delta + \frac{\Delta t}{l}\right)\right]
%     \bigg\}\ ,
  \end{align*}
  where $\Delta t := t - t'$, $\gamma_D = \frac{D l}{2}$.
  Similarly the covariance of target mRNAs is
  \begin{multline*}
    k_{m_j m_k}(t, t')
    = \frac{\sqrt{\pi} l a^2 S_j S_k}{2} \bigg(
    h'_{jk}(t, t', \delta) + h'_{kj}(t', t, \delta) \\
    - h'_{jk}(t, t', D_j) - h'_{kj}(t', t, D_k)
    \bigg)\ ,
  \end{multline*}
  where
  \begin{multline*}
    h'_{jk}(t, t', D_x) = 
    \frac{\exp(\gamma_x^2 -D_x t - D_k t')}{(D_x + \delta) (D_j - \delta)}
    \bigg\{ 
    \\
    % \frac{(D_k + \delta)\exp((D_k-\delta) t') - 2\delta}{(D_k^2-\delta^2)}
    \left(\frac{\exp((D_k-\delta) t') - 1}{D_k-\delta} +
      \frac{1}{D_k + D_x} \right)
    \left[\erf\left(\gamma_x - \frac{t}{l}\right) - \erf\left(\gamma_x\right)\right]
    \\
    + \frac{\exp((D_k+D_x)t')}{D_k+D_x}
    \left[\erf\left(\gamma_x + \frac{t'}{l}\right)
    - \erf\left(\gamma_x - \frac{\Delta t}{l}\right)\right]
    \bigg\}\ ,
  \end{multline*}
  where $\gamma_x = \frac{D_x l}{2}$.
  Full model for inference of $p(t)$ is derived in Text S1.

  \section{Training set ranking}
  The training set or multiple-target ranking was formed by fitting
  joint models to five assumed known targets together with each new
  candidate independently and ranking by model likelihood.  In our
  approach, the training set was formed by taking 5 highest-ranking
  targets from the single-target method, but naturally any existing background
  knowledge of known targets should be used.  To speed up the
  computation, we first fitted a Gaussian process model to the
  training set genes alone.  All the parameters concerning these genes
  were then fixed when fitting the larger models to the training set
  plus each candidate.

  \section{In-situ and knock-out data processing}
  Genes with any of the annotation terms listed in Supplementary Table
  3 of~\cite{Zinzen2009} in the data of~\cite{Tomancak2002} were
  considered to be expressed in mesoderm or muscle tissue.
  The knock-out data was processed as explained
  in~\cite{Sandmann2006a}.  $q$-values were computed at each time
  point using SAM~\cite{Saeed2003,Tusher2001}, the values at
  neighbouring time points were averaged and the minimum of these
  averages was used as the final value.  A cut-off value of 0.1 was
  used for detecting significant differential expression as
  in~\cite{Sandmann2006a}.

\end{materials}

\begin{acknowledgments}
A.H. was supported by Postdoctoral Researcher's Project No 121179 of the Academy of Finland.
M.R. and N.D.L acknowledge support from EPSRC Grant No EP/F005687/1 "Gaussian Processes for Systems Identification with Applications in Systems Biology". 
This work was supported in part by the IST Programme of the European Community, under the PASCAL2 Network of Excellence, IST-2007-216886. This publication only reflects the authors' views.
\end{acknowledgments}

%\bibliographystyle{pnas}
%\bibliography{disim}

\begin{thebibliography}{10}

\bibitem{Ren2000}
Ren, B et~al.
\newblock (2000) Genome-wide location and function of {DNA} binding proteins.
\newblock \emph{Science} 290:2306--2309.

\bibitem{Iyer2001}
Iyer, VR et~al.
\newblock (2001) Genomic binding sites of the yeast cell-cycle transcription
  factors {SBF} and {MBF}.
\newblock \emph{Nature} 409:533--538.

\bibitem{Li2008}
Li, X et~al.
\newblock (2008) Transcription factors bind thousands of active and inactive
  regions in the {D}rosophila blastoderm.
\newblock \emph{PLoS Biol} 6:e27.

\bibitem{MacArthur2009}
MacArthur, S et~al.
\newblock (2009) Developmental roles of 21 {D}rosophila transcription factors
  are determined by quantitative differences in binding to an overlapping set
  of thousands of genomic regions.
\newblock \emph{Genome Biol} 10:R80.

\bibitem{Furlong2001}
Furlong, EE, Andersen, EC, Null, B, White, KP, Scott, MP
\newblock (2001) Patterns of gene expression during {D}rosophila mesoderm
  development.
\newblock \emph{Science} 293:1629--1633.

\bibitem{Sandmann2006a}
Sandmann, T et~al.
\newblock (2006) A temporal map of transcription factor activity: {M}ef2
  directly regulates target genes at all stages of muscle development.
\newblock \emph{Dev Cell} 10:797--807.

\bibitem{Gitter2009}
Gitter, A et~al.
\newblock (2009) Backup in gene regulatory networks explains differences
  between binding and knockout results.
\newblock \emph{Mol Syst Biol} 5:276.

\bibitem{Gao2008}
Gao, P, Honkela, A, Rattray, M, Lawrence, ND
\newblock (2008) Gaussian process modelling of latent chemical species:
  applications to inferring transcription factor activities.
\newblock \emph{Bioinformatics} 24:i70--i75.

\bibitem{Rasmussen2006}
Rasmussen, CE, Williams, CKI
\newblock (2006) \emph{Gaussian Processes for Machine Learning}
\newblock (MIT Press).

\bibitem{Lawrence2010}
Lawrence, ND, Rattray, M, Gao, P, Titsias, M
\newblock (2010) in \emph{Learning and Inference in Computational Systems
  Biology}, eds{} Lawrence, ND, Girolami, M, Rattray, M, Sanguinetti, G
\newblock (MIT Press, Cambridge, MA).

\bibitem{Bansal2007a}
Bansal, M, Belcastro, V, Ambesi-Impiombato, A, di~Bernardo, D
\newblock (2007) How to infer gene networks from expression profiles.
\newblock \emph{Mol Syst Biol} 3:78.

\bibitem{Ernst2005}
Ernst, J, Nau, GJ, Bar-Joseph, Z
\newblock (2005) Clustering short time series gene expression data.
\newblock \emph{Bioinformatics} 21 Suppl 1:i159--i168.

\bibitem{Stolovitzky2007}
Stolovitzky, G, Monroe, D, Califano, A
\newblock (2007) Dialogue on reverse-engineering assessment and methods: the
  {DREAM} of high-throughput pathway inference.
\newblock \emph{Ann N Y Acad Sci} 1115:1--22.

\bibitem{Barenco2006a}
Barenco, M et~al.
\newblock (2006) Ranked prediction of p53 targets using hidden variable dynamic
  modeling.
\newblock \emph{Genome Biol} 7:R25.

\bibitem{Gatta2008}
{Della Gatta}, G et~al.
\newblock (2008) Direct targets of the {TRP63} transcription factor revealed by
  a combination of gene expression profiling and reverse engineering.
\newblock \emph{Genome Res} 18:939--948.

\bibitem{Tomancak2002}
Tomancak, P et~al.
\newblock (2002) Systematic determination of patterns of gene expression during
  {D}rosophila embryogenesis.
\newblock \emph{Genome Biol} 3:RESEARCH0088.

\bibitem{Liu2005}
Liu, X, Milo, M, Lawrence, ND, Rattray, M
\newblock (2005) A tractable probabilistic model for {A}ffymetrix probe-level
  analysis across multiple chips.
\newblock \emph{Bioinformatics} 21:3637--3644.

\bibitem{Pearson2009}
Pearson, RD et~al.
\newblock (2009) puma: a {B}ioconductor package for propagating uncertainty in
  microarray analysis.
\newblock \emph{BMC Bioinformatics} 10:211.

\bibitem{Khanin2006}
Khanin, R, Vinciotti, V, Wit, E
\newblock (2006) Reconstructing repressor protein levels from expression of
  gene targets in {E}scherichia coli.
\newblock \emph{Proc Natl Acad Sci U S A} 103:18592--18596.

\bibitem{Sandmann2007}
Sandmann, T et~al.
\newblock (2007) A core transcriptional network for early mesoderm development
  in {D}rosophila melanogaster.
\newblock \emph{Genes Dev} 21:436--449.

\bibitem{Zinzen2009}
Zinzen, RP, Girardot, C, Gagneur, J, Braun, M, Furlong, EEM
\newblock (2009) Combinatorial binding predicts spatio-temporal cis-regulatory
  activity.
\newblock \emph{Nature} 462:65--70.

\bibitem{Barenco2009}
Barenco, M et~al.
\newblock (2009) {rHVDM}: an {R} package to predict the activity and targets of
  a transcription factor.
\newblock \emph{Bioinformatics} 25:419--420.

\bibitem{Saeed2003}
Saeed, AI et~al.
\newblock (2003) {TM4}: a free, open-source system for microarray data
  management and analysis.
\newblock \emph{Biotechniques} 34:374--378.

\bibitem{Tusher2001}
Tusher, VG, Tibshirani, R, Chu, G
\newblock (2001) Significance analysis of microarrays applied to the ionizing
  radiation response.
\newblock \emph{Proc Natl Acad Sci U S A} 98:5116--5121.

\end{thebibliography}

%\begin{thebibliography}{10}
%\bibliography{puma_package}
%\end{thebibliography}
\end{article}

% Figures


\end{document}


